packages/office-ui-fabric-react/src/components/List/List.tsx

```diff
@@ -369,7 +369,12 @@ export class List extends BaseComponent<IListProps, IListState> implements IList
     }
 
     return (
-      <div ref={this._root} {...divProps} role={role === undefined ? 'list' : role} className={css('ms-List', className)}>
+      <div
+        ref={this._root}
+        {...divProps}
+        role={role === undefined ? (pageElements.length === 0 ? '' : 'list') : role}
```

Comment:
```suggestion
        role={role === undefined ? (pageElements.length === 0 ? undefined : 'list') : role}
```

This way we don't have an empty and invalid `role` in the DOM in the case of an empty `List`. `undefined` tells React to not render the attribute on the DOM.

Tho it would be great if we could _somehow_ avoid the nested in-line ternary as its quite difficult to parse as a reader.